### DISCUSSION TITLE
How to use listing categories and shortcodes with project profiles

### DISCUSSION THREAD

**Discussion by timothoms** [quarto-dev/quarto-cli#8414] [Q&A]
### Description

We have used Quarto project profiles to create a relatively complex bilingual website, and for the most part this has worked very well. We use language-specific metadata files (e.g. \_metadata-fr.yml) which are called in the profile yaml (\_quarto-fr.yml) for document metadata such as page titles, and then we create language-specific versions of any given page from the same .qmd file by including profile content with: 

```
::: {.content-visible when-profile="fr"}

:::
```

I have two questions about functionality that we could not accomplish with profiles. 

First, we are not able to translate listing categories. We were able to translate certain parts of the categories sidebar with \_language.yml, but not the categories which we defined. I thought it should be possible to assign categories in the language-specific metadata files and then include them in the .qmd frontmatter with shortcodes but the category sidebar shows the shortcodes themselves rather than the categories (similar to this issue except that we use meta rather than var shortcodes: https://github.com/quarto-dev/quarto-cli/issues/5578#issuecomment-1553846352). Is there another possible approach for translating listing categories? 

Second, we have a sidebar for country pages which we would like to order alphabetically for the language shown. The items included in the sidebar are created automatically from the .qmd files in specific folders, and because these .qmd files have the English country names in their file names, they get sorted correctly in the sidebar for the English site. But we create the French versions of these pages from the same .qmd files, and the French sidebar has the translated country names (from \_metadata-fr.yml) but in the order of the English names. 

I am aware of the 'order' option which can be supplied in the document frontmatter, but again I would need to read this from profile metadata with a shortcode, and I have not been able to make this work. Here is a sample error message: 

```
ERROR: In file quarto-site-generator/Africa/SouthAfrica.qmd
(line 6, columns 8--43) Field "order" has value {{< meta ctry_order.SouthAfrica >}}, which must instead be a number
5: title: "{{< meta ctry_title.SouthAfrica >}}"
6: order: {{< meta ctry_order.SouthAfrica >}}
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
7: execute: 
✖ The value {{< meta ctry_order.SouthAfrica >}} is of type object.
ℹ The error happened in location order.
```

Both of these issues may be because I don't know enough about how and when shortcodes work. It seems that they only work in document frontmatter when in quotation marks -- I also tried without -- but then yaml interprets their content as character strings, which doesn't work for all options. Is there documentation on which metadata fields can use shortcodes?

Thank you for any pointers you may be able to provide. 

Quarto version: 1.3.450
OS: macOS 14.3

**Comment by mcanouil** [comment]
Would you mind sharing a small Git repository with the setup and highlight the part that does not work?

**Reply by timothoms** [reply]
Here is a small repository with both issues: [https://github.com/tjetdata/sandbox](https://github.com/tjetdata/sandbox)

As is, it will halt the render with an error message. With the order field commented out in the 4 country pages, it will run and show the category sidebar issue on the Documents page. Many thanks for taking a look!

**Comment by timothoms** [comment]
Since the second issue can be manually resolved -- by simply listing all the individual pages in the sidebar navigation in the correct order without the auto option -- I changed the above-mentioned [repository](https://github.com/tjetdata/sandbox) to only feature the first issue of translating listing categories with profiles. See the screenshots below for the problem. 

I understand from [issue # 5578](https://github.com/quarto-dev/quarto-cli/issues/5578#issuecomment-1553846352) that there is no simple solution but I still wonder if there is any undocumented workaround. 

![Screenshot 2024-02-05 at 12 58 10 PM](https://github.com/quarto-dev/quarto-cli/assets/20423161/d485a8b6-07d9-4e9b-abff-e5048d16b113)
![Screenshot 2024-02-05 at 12 58 18 PM](https://github.com/quarto-dev/quarto-cli/assets/20423161/509d036d-1497-41e6-8985-6a137c0b77ea)

**Reply by mcanouil** [reply]
After some experiments, I don't see a solution/workaround right now.

@dragonstyle Do you know if it's possible to evaluate the shortcodes before retrieving the categories to make the listings?

**Reply by dragonstyle** [reply]
@cscheid will also want to weigh in to be sure that I'm right about this, but the problem right now is that shortcodes are evaluated in LUA (when pandoc rendering pages), but the categories are read before pandoc rendering the documents. We workaround this in some other places by waiting to emplace the content until after the rendering has taken place (so we can use the rendered version) but we aren't doing this with categories.

I also don't see a great workaround to this right now.

**Reply by mcanouil** [reply]
In might go to/in:
- https://github.com/quarto-dev/quarto-cli/issues/6518
